pkg/kubectl/cmd/util/factory.go

```diff
@@ -254,8 +254,8 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return kubectl.ReaperFor(mapping.Kind, client, expClient)
 		},
-		Validator: func() (validation.Schema, error) {
-			if flags.Lookup("validate").Value.String() == "true" {
+		Validator: func(validate bool) (validation.Schema, error) {
```

Comment:
Since `--validate` only makes sense for certain commands, I removed it from global flags, and therefore now this flag is in `cmd.Flags()` instead of `factory.flags`. This means to check the value of this flag, we should use `cmd.Flags().GetBool()` instead of `flags.Lookup()`. Passing the bool value here is better than passing `cmd`, imo. If `Validator()` is called with `validate == false`, `validation.NullSchema{}` will be returned.